### DISCUSSION TITLE
Skipping Overriden Keys

### DISCUSSION THREAD

**Discussion by mrxinu** [tkainrad/keycombiner#72] [Q&A] [answered]
When I was going through the Chrome collection I noticed I was only able to practice with 2/5 keys because I was in a web browser. That's fine, but it also prevents me from progressing beyond the first level because I can never master 5/5.

I know that the desktop app would probably get me around that but I work for a security company and I'd have to put the app through an entire process to install it. So, can't do that just yet.

**Comment by tkainrad** [comment, marked as answer]
Hi, thanks for starting this topic!

Which behavior would you rather see? Just not skipping the combinations isn't really an option as your web browser will not forward these combinations to KeyCombiner. Practicing them will therefore not work.

Happy to discuss other possibilities. For now, there'd be two workarounds:

1. Complete the first lesson on a personal machine where you can install the desktop app. You should then be able to complete the _Search_ lesson in the browser and subsequently, all lessons/modules should be possible.
2. Collect the combinations of later modules into a personal collection of your own and just practice the Chrome collection this way. You can still use the course system for other applications.

**Reply by mrxinu** [reply]
You're very welcome! Thanks for the quick reply!

With the lesson progress being shared across the web & desktop versions I can't think of any clean way to keep the gamified progression through the levels. Removing that blocker so that all lessons can be accessed without mastering previous ones would be interesting. Then again, it still gamified if you can just skip forward?

Thanks for the workaround suggestions!

1. I don't do enough non-work stuff to justify booting up this loud gaming rig unless I'm settling in for a weekend of gaming. 😆 
2. I gave this a try and it does create personal collections with the name of the lesson (e.g., Editing: Lines from the VSCode collection) but it doesn't say which collection it's from (could be editing lines from any application) so I feel like that would get confusing quickly.

I'll have to give it some more thought, but for now I'd say leaving everything as it is right now makes the most sense!

**Reply by tkainrad** [reply]
Glad that you are sort of happy with the status quo :sweat_smile: 

> I gave this a try and it does create personal collections with the name of the lesson (e.g., Editing: Lines from the VSCode collection) but it doesn't say which collection it's from (could be editing lines from any application) so I feel like that would get confusing quickly.

You could rename the created personal collections to whatever you want if you wanted to use the collection name to remember which lesson it was copied from. 

However, I'd even suggest collecting combinations from multiple/all Chrome lessons into the same personal collection. KeyCombiner has a [_Context_](https://github.com/tkainrad/keycombiner/wiki/Context-VS-Category) field so that you never forget which application (or more general _Context_) a combination belongs to. Any combination from the [VSCode course](https://keycombiner.com/courses/master-vscode/) and also [public collection](https://keycombiner.com/collections/vscode/) will have _VSCode_ as _Context_. Personally, I even mix up different applications into the same collections. Using the quick filter panels on the right side, it's then still easy to filter by _Context_/application.

That being said, I will still think about restructuring the _Chrome_ course a bit so that there is at least one lesson in all modules that can be mastered via Browser.

**Reply by darthwalsh** [reply]
@tkainrad  i've seen that PWAs (browser apps you have installed, running in their own window) can capture keyboard shortcuts that the browser doesn't let normal website tabs handle. 

I thought I could find more information by googling, but I'm not finding the right documentation... here's what I've found so far:

- https://bugs.chromium.org/p/chromium/issues/detail?id=119881#c84
    - this chromium request is about letting websites handle browser-shortcuts, which wasn't implemented because standalone PWAs can handle them, see comment 84
- https://github.com/microsoft/vscode/issues/150735
    - the vscode PWA handles `Ctrl-W` or `Ctrl-N`

It's a bit of extra work to implement a PWA, when you have an already working website and desktop app, but this might interest you? :)
